### ISSUE TITLE
Proposals for improvement

### ISSUE THREAD

**Issue by AlejandroUPC** [arthurprevot/yaetos#69]
Hey, really enjoyed going through the code and  it's a great tool!

I have some proposals that in my opinion might bring some value for the project:

- Remove all the prints for actual logs
- In order to have some standardization in the code I would propose black  as a linter(personal preference).
- Typing on every function.
- Use context managers when possible(saw some .open() .close() around, which is not an issue but rather use the builtins from python and looks more elegant in my personal opinion)
- Use mypy (found like 2-3 syntax errors when I ran mypy on the project).

**Comment by arthurprevot** [comment]
Thanks a lot @AlejandroUPC for the suggestions. They look all good to me in general. More details below:

- Remove all the prints for actual logs -> 👍 
- In order to have some standardization in the code I would propose black (personal preference). -> I would like to add linting to make it consistent. black looks a bit too spread to me, but definitely in on making it more consistent (pep8)
- Typing on every function. -> agreed 
- Use context managers when possible(saw some .open() .close() around, which is not an issue but rather use the builtins from python and looks more elegant in my personal opinion) -> are you talking about using `with` statements ? If so, yes !
- Use mypy (found like 2-3 syntax errors when I ran mypy on the project). -> not familiar with mypy, I will check it out.

**Comment by arthurprevot** [comment]
By the way, I just merged some changes today that provide better interfaces to run the code (using `yaetos` commandlines, like `yaetos run_dockerized some/job.py --some=args`, or using jupyter), you may want to check it out at  https://github.com/arthurprevot/yaetos/pull/67 .

**Comment by AlejandroUPC** [comment]
Thanks will check it out, I am going to push some changes in the same branch, feel free to act on them.

Also I saw in the config files some credentials are stored on config files on plain text, I am not a big fan of it and rather somewhat environment variables (credentials still have to be in one file, but just one instead of one per each config)

**Comment by arthurprevot** [comment]
Thanks. Regarding the updates in https://github.com/arthurprevot/yaetos/pull/70, thanks, I put some details there.

Regarding what to focus on, it is pretty open and up to you. What you defined up here are good things to tackle. I would tackle them in that order:
- [x] basic pep8 compliance + automated checks in CI: done in last 3 PRs. I disabled some checks but I think the important ones are in.
- [ ] Typing on every function + automated checks in CI (mypy)
- [ ] Remove all the prints for actual logs
- [ ] Use context managers when possible(saw some .open() .close() around, which is not an issue but rather use the builtins from python and looks more elegant in my personal opinion)

There are other potential improvements listed at the bottom of https://github.com/arthurprevot/yaetos page.

What part of the code is not clean or not well maintained:
 * oracle related code. I don't have access to an oracle instance anymore and it is not a priority for me. It could be removed from the code.
 * koalas related code. I disabled it but I would like to re-integrate it eventually, after fixing some dependency issues. Also, koalas changed in the latest version of spark, so it would ideally come from the latest version (if available in EMR, and in AWS Data Pipeline)
 * code related to running pandas 100% jobs (no spark involved). I added that recently. It works but it is not well integrated yet. I added 2 files, (`spark_utils.py` and `pandas_utils.py`) and ideally all engine specific code should be in there, and not in `etl_utils.py`, but it is not the case right now.
 * `Dockerfile*` and `requirement*.txt` setup. Messy. There are both python2.7 and python3 setups in parallel. python2 could be dropped. Also, some job specific libraries should be removed from the main requirement.txt file (salesforce, stripe libs for ex).
 * example jobs: it could use some cleaning.

**Comment by AlejandroUPC** [comment]
Hey its been 2 years, might have some time in my hands the next few weeks, anything you need help with/roadmap?

**Comment by arthurprevot** [comment]
Hi @AlejandroUPC ! Good to hear from you. 
Yes, there are some things I am implementing for a use case at a client (sending jobs to kubernetes instead of EMR) and other things I would like to add/improve (a genAI interface to the data, a basic UI...). I can list more. Any help is welcome !
 
There were a bunch of updates since we spoke: integration of airflow, integration with "Panel" for dashboarding, support for single tenant architecture, addition of a bunch of use cases (around gen AI, image processing ... ), etc. Some of the code was implemented quickly and could be improved.

**Comment by AlejandroUPC** [comment]
old mr no longer relevant
